src/biconnectivity13/biconnect.jl

```diff
@@ -84,16 +27,72 @@ julia> biconnected_components(cycle_graph(5))
 """
 function biconnected_components end
 @traitfn function biconnected_components(g::::(!IsDirected))
-    state = Biconnections(g)
-    for u in vertices(g)
-        if state.depth[u] == 0
-            visit!(g, state, u, u)
-        end
+        n = nv(g)
+        S = Array{Tuple{Int, Int}, 1}(undef,n)
```

Comment:
use `Vector` instead of `Array{..., 1}`.

but if `g` is disconnected, won't this overallocate?

Also, why `Int` here? (see below)

Reply:
As it turns out from my experiments, it won’t make a difference.

Reply:
yes, i will do it